addons/sale_stock/models/stock_config_settings.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class StockConfigSettings(models.TransientModel):
+    _inherit = 'stock.config.settings'
+
+    security_lead = fields.Float(related='company_id.security_lead', default=lambda self: self.env.user.company_id.security_lead, string="Security Lead Time for Sales")
```

Comment:
A related is just some specific computed field and should therefore not have a default value.